SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -51,29 +58,90 @@ QueryConstraint InterestFactory::CreateClientCheckoutRadiusConstraint(USpatialCl
 
-	// Get interest distances for each actor.
-	TMap<UClass*, float> ActorComponentSetToRadius = CheckoutRadiusConstraintUtils::GetActorTypeToRadius();
+		// Get interest distances for each actor.
+		TMap<UClass*, float> ActorComponentSetToRadius = CheckoutRadiusConstraintUtils::GetActorTypeToRadius();
 
-	// For every interest distance that we still want, build a map from radius to list of actor type components that match that radius.
-	TMap<float, TArray<UClass*>> DistanceToActorTypeComponents = CheckoutRadiusConstraintUtils::DedupeDistancesAcrossActorTypes(
-		ActorComponentSetToRadius);
+		// For every interest distance that we still want, build a map from radius to list of actor type components that match that radius.
+		TMap<float, TArray<UClass*>> DistanceToActorTypeComponents = CheckoutRadiusConstraintUtils::DedupeDistancesAcrossActorTypes(
+			ActorComponentSetToRadius);
 
-	// The previously built map dedupes spatial constraints. Now the actual query constraints can be built of the form:
-	// OR(AND(cyl(radius), OR(actor 1 components, actor 2 components, ...)), ...)
-	// which is equivalent to having a separate spatial query for each actor type if the radius is the same.
-	TArray<QueryConstraint> CheckoutRadiusConstraints = CheckoutRadiusConstraintUtils::BuildNonDefaultActorCheckoutConstraints(
-		DistanceToActorTypeComponents, ClassInfoManager);
+		// The previously built map dedupes spatial constraints. Now the actual query constraints can be built of the form:
+		// OR(AND(cyl(radius), OR(actor 1 components, actor 2 components, ...)), ...)
+		// which is equivalent to having a separate spatial query for each actor type if the radius is the same.
+		TArray<QueryConstraint> CheckoutRadiusConstraints = CheckoutRadiusConstraintUtils::BuildNonDefaultActorCheckoutConstraints(
+			DistanceToActorTypeComponents, ClassInfoManager);
 
-	// Add all the different actor queries to the overall checkout constraint.
-	for (auto& ActorCheckoutConstraint : CheckoutRadiusConstraints)
+		// Add all the different actor queries to the overall checkout constraint.
+		for (auto& ActorCheckoutConstraint : CheckoutRadiusConstraints)
+		{
+			CheckoutRadiusConstraintRoot.OrConstraint.Add(ActorCheckoutConstraint);
+		}
+	}
+	else
 	{
-		CheckoutRadiusConstraint.OrConstraint.Add(ActorCheckoutConstraint);
+		const TArray<float> NetCullDistances = ClassInfoManager->GetNetCullDistances();
+
+		for (float NetCullDistance : NetCullDistances)
+		{
+			const float MaxCheckoutRadiusMeters = CheckoutRadiusConstraintUtils::NetCullDistanceSquaredToSpatialDistance(NetCullDistance);
+			const uint32 NCDComponentId = ClassInfoManager->GetComponentIdForNetCullDistance(NetCullDistance);
+
+			QueryConstraint ComponentConstraint;
+			ComponentConstraint.ComponentConstraint = NCDComponentId;
+
+			if (SpatialGDKSettings->bEnableNetCullDistanceFrequency)
+			{
+				float FullFrequencyCheckoutRadius = MaxCheckoutRadiusMeters * SpatialGDKSettings->FullFrequencyNetCullDistanceRatio;
```

Comment:
This introduces a dependency between this "max" distance to how frequency is configured, which IIUC was somewhat of a placeholder hack until larger radiuses were handled efficiently by the runtime. What is the plan for this once this max is no longer a thing?

Reply:
I think it might be better if we give users a default "min radius" to configure rather than a ratio to the max distance

Reply:
Not quite, this is the max WRT the native checkout radius, which can be modified on a per-class basis.